### PR TITLE
Team partial membership

### LINKED ISSUE #457: Team partial membership
Should be able to set percentage of team involvement. 

1. Add a column 'percentage' to groupMembership (which should be optional)
2. Change 'Add to team' flow - instead of a combobox create a dialog allowing setting percentage and roles
3. When adding, chek user not to allow more than 100% involvement

### PR DESCRIPTION
**Pull request by tsumo**
## PR includes

-   [x] Feature

## Related issues

Resolve #457 

Screenshot 2024-02-12 at 10 42 35

Screenshot 2024-02-12 at 10 42 23

### CODE REVIEW

**Review by 9teen90nine — APPROVED** [review]

File: src/components/AddUserToTeamForm/AddUserToTeamForm.tsx [hunk 1]
```diff
@@ -1,89 +1,138 @@
...
+
+    const percentage = watch('percentage');
+
+    useEffect(() => {
+        if (percentage && percentage > max) {
+            setError('percentage', { message: tr('Maximum value is {max}', { max }) });
+        } else {
+            clearErrors('percentage');
+        }
+    }, [max, percentage, setError, clearErrors]);
```
Comment (DenisVorop) [inline src/components/AddUserToTeamForm/AddUserToTeamForm.tsx:84]: maybe would be better to create percentage handler, that will have condition resolver instead useEffect hook, ex

```ts
onPercentageHandler = (value) => {
    setValue(value)
   
   if (condition) {
        etc...
    }
}
```

File: src/modules/userMethods.ts [hunk 2]
```diff
@@ -239,6 +248,12 @@ export const userMethods = {
         });
     },
 
+    getAvailableMembeshipPercentage: async (userId: string): Promise<number> => {
```
Comment (DenisVorop) [inline src/modules/userMethods.ts:251]: `getAvailableMembershipPercentage`

File: src/components/AddUserToTeamForm/AddUserToTeamForm.tsx (hunk 1 above)
Comment (tsumo) [reply src/components/AddUserToTeamForm/AddUserToTeamForm.tsx]: done

File: src/modules/userMethods.ts (hunk 2 above)
Comment (tsumo) [reply src/modules/userMethods.ts]: done

**Review by DenisVorop — APPROVED** [review]

**Pull request merged**